apps/fz_http/lib/fz_http_web/live/mfa/register_steps_component.ex

```diff
@@ -0,0 +1,132 @@
+defmodule FzHttpWeb.MFA.RegisterStepsComponent do
+  @moduledoc """
+  MFA registration steps
+  """
+  use Phoenix.Component
+
+  import FzHttpWeb.ErrorHelpers
+
+  def render_step(assigns) do
+    case assigns[:step] do
+      :pick_type ->
+        pick_type(assigns)
+
+      :register ->
+        register(assigns)
+
+      :verify ->
+        verify(assigns)
+
+      :save ->
+        save(assigns)
+    end
+  end
+
+  def pick_type(assigns) do
+    ~H"""
+    <form id="mfa-method-form" phx-target={@parent} phx-submit="next">
+      <h4>Choose authenticator type</h4>
+      <hr>
+
+      <div class="control">
+        <div>
+          <label class="radio">
+            <input type="radio" name="type" value="totp" checked>
+            Time-Based One-Time Password
+          </label>
+        </div>
+        <div>
+          <label class="radio disabled">
+            <input type="radio" name="type" value="native" disabled>
+            Native (Windows Hello, iOS Face ID, etc)
+          </label>
+        </div>
+        <div>
+          <label class="radio disabled">
+            <input type="radio" name="type" value="portable" disabled>
+            Portable (YubiKey-like products)
+          </label>
+        </div>
+      </div>
+    </form>
+    """
+  end
+
+  def register(assigns) do
+    secret = NimbleTOTP.secret()
+
+    assigns =
+      Map.merge(
+        assigns,
+        %{
+          secret: secret,
+          secret_base64_encoded: Base.encode64(secret),
+          uri: NimbleTOTP.otpauth_uri("Firezone:test@test.com", secret, issuer: "Firezone")
```

Comment:
This will change to retrieve the current user later on for the label, right?

Reply:
Yup, will clean this up Monday. This is not quite ready for merging yet. Will need to explore how to make this testable.

And I think there is probably not too much work left to integrate this into the auth flow. I might get this done in the same PR.